FinancialTransactionsApi/V1/UseCase/GetAllUseCase.cs

```diff
@@ -21,21 +20,9 @@ public GetAllUseCase(ITransactionGateway gateway)
 
         public async Task<PaginatedResponse<TransactionResponse>> ExecuteAsync(TransactionQuery query)
         {
-            var transactions = await _gateway.GetPagedTransactionsAsync(query).ConfigureAwait(false);
-
-            var page = query.Page > 1 ? query.Page : 0;
-
-            var filteredTransactions = transactions.Skip(page * query.PageSize).Take(query.PageSize);
-
-            var paginateTransactions = new Paginated<Transaction>()
-            {
-                Results = filteredTransactions,
-                CurrentPage = page,
-                PageSize = query.PageSize,
-                TotalResultCount = transactions.Count(),
-            };
+            Paginated<Transaction> result = await _gateway.GetPagedTransactionsAsync(query).ConfigureAwait(false);
```

Comment:
I am unsure about the decision to remove the Pagination code from the UseCase layer
In my opinion, the Gateway classes (essentially the repository layer) should just retrieve the data from the database (No pagination)
And the UseCase classes (the domain layer) should be doing all the Pagination.

Am happy to discuss and be convinced otherwise